python/pyspark/mllib/tests.py

```diff
@@ -134,6 +134,14 @@ def test_sparse_vector_indexing(self):
         for ind in [4, -5, 7.8]:
             self.assertRaises(ValueError, sv.__getitem__, ind)
 
+    def test_matrix_indexing(self):
+        mat = DenseMatrix(3, 2, [0, 1, 4, 6, 8, 10])
+        numcols = mat.numCols
+        for ind, val in enumerate(mat.values):
```

Comment:
I think it should be okay to write down the checks explicitly. Or you can do this:

~~~python
expected = [[0, 6], [1, 8], [4, 10]]
for i in range(3):
  for j in range(2):
    self.assertEquals(mat[i, j], xpected[i][j])
~~~